### 🌐 API(Application Programming Interface)란 무엇인가?
#### 1. API의 정의
* **API**는 **Application Programming Interface**의 약자로, 서로 다른 소프트웨어 프로그램이나 시스템이 **서로 데이터를 주고받을 수 있게 연결해 주는 창구(매개체)**입니다.
* 데이터베이스나 서버에 직접 접근하지 않고, 규격화된 약속(인터페이스)을 통해 안전하게 필요한 데이터만 요청하고 응답받을 수 있습니다.
---
#### 2. API를 이해하는 가장 쉬운 비유: 🍽️ 식당의 점원 (Waiter)
| 구분 | 역할 비유 | 실제 컴퓨터 환경 |
| :--- | :--- | :--- |
| **손님 (Client)** | 메뉴를 보고 음식을 주문함 | **내 컴퓨터 / 파이썬 프로그램 / 브라우저** |
| **점원 (API)** | 주문을 주방에 전달하고, 음식을 손님에게 가져다줌 | **API (Application Programming Interface)** |
| **주방 (Server)** | 데이터(음식)를 보관하고 요리(처리)함 | **원격 데이터베이스 / 서버** |
> 💡 우리가 직접 주방(서버)에 들어가지 않아도, **점원(API)**에게 올바른 메뉴(요청)를 말하면 정해진 방식(응답)으로 음식(데이터)을 받아볼 수 있습니다.
---
#### 3. 웹 API의 핵심 구조: 요청(Request)과 응답(Response)
API와의 통신은 크게 **두 가지 단계**로 이루어집니다.
1. **요청 (Request)**: 클라이언트가 서버에게 특정 데이터를 달라고 보냄
   * **URL (주소)**: 데이터를 요청할 위치
   * **인증키 (API Key)**: 접근 권한 확인용 열쇠
   * **파라미터 (Parameter)**: 원하는 데이터 조건 (예: 연도, 자치구, 개수 등)
2. **응답 (Response)**: 서버가 처리 결과를 클라이언트에게 전달함
   * 주로 **JSON** 또는 **XML** 형식을 사용합니다.
---
#### 4. 데이터 교환의 표준 형식: JSON (JavaScript Object Notation)
현재 대부분의 웹 API는 데이터를 **JSON** 형식으로 제공합니다.
* Python의 **사전(Dictionary)** 구조(`{ Key : Value }`)와 매우 유사합니다.
* 읽고 쓰기 쉽고 가벼워서, 파이썬의 `pandas` 라이브러리를 통해 **표(DataFrame)** 형태로 손쉽게 변환할 수 있습니다.
```json
{
  "name": "홍길동",
  "age": 25,
  "city": "서울"
}

### 🔑 서울 열린데이터 광장 API 인증키 발급 안내
서울시 오픈 API 데이터(1회 1,000건 한도) 수집을 위한 **개인 인증키 발급 절차**입니다.
#### 1. 회원가입 및 로그인
1. [서울 열린데이터 광장 홈페이지](https://data.seoul.go.kr) 접속
2. 우측 상단의 **[회원가입]** 진행 후 **[로그인]**
#### 2. 데이터셋 검색 및 OpenAPI 신청
1. 검색창에 데이터명 검색 (예: `서울시 부동산 전월세가 정보`)
2. 검색 결과 클릭 후 우측 상단의 **[OpenAPI 신청]** 버튼 클릭
#### 3. 인증키 신청 양식 작성
* **활용 용도**: `교육/연구` 또는 `개인 데이터 분석`
* **URL**: `http://localhost` (개인 웹사이트가 없는 경우)
* 약관 동의 후 **[신청]** 클릭 *(즉시 자동 승인/발급)*
#### 4. 발급된 인증키 확인
1. 우측 상단 **[마이페이지]** $\rightarrow$ **[인증키 관리]** 이동
2. 발급된 **인증키 (32자리 문자열)** 확인 및 복사
#### 5. 인증키 보안 설정 (`.env` 파일 저장)
인증키는 소스코드에 직접 노출하지 않고 `.env` 파일에 저장하여 관리합니다.
##### `.env` 파일 작성 예시:
```env
SEOUL_API_KEY=your_actual_32digit_api_key_here

**열린데이터광장 API키는 위험성이 크지 않지만, 과금하는 API키를 코드에 직접 넣어서 노출하는 것은 "내돈을 가져가세요"라고 하는것과 같습니다. 조심 또 조심!**


In [1]:
# 제 API 키입니다. 각자 발급받은 키를 사용하세요. 
# 이해를 돕기위해 일단 코드에 직접 넣었습니다.

SEOUL_API_KEY='584f61755773657535326b707a6253'

In [2]:
import requests
import pandas as pd
# 1. API 설정
API_KEY = SEOUL_API_KEY  # 실제 사용 시 발급받은 인증키로 변경하세요.
TYPE = "json"
SERVICE = "tbLnOpendataRentV"
START_INDEX = 1
END_INDEX = 5
# 2. 요청 URL 생성
url = f"http://openapi.seoul.go.kr:8088/{API_KEY}/{TYPE}/{SERVICE}/{START_INDEX}/{END_INDEX}/"
# 3. API 호출
response = requests.get(url)
data = response.json()
# 4. DataFrame 변환
df = pd.DataFrame(data[SERVICE]['row'])
# 5. 결과 확인
df.head()

,RCPT_YR,CGG_CD,CGG_NM,STDG_CD,STDG_NM,LOTNO_SE,LOTNO_SE_NM,MNO,SNO,FLR,...,GRFE,RTFE,BLDG_NM,ARCH_YR,BLDG_USG,CTRT_PRD,NEW_UPDT_YN,CTRT_UPDT_USE_YN,BFR_GRFE,BFR_RTFE
0,2026,11260,중랑구,10100,면목동,1,대지,1112,0000,5.0,...,11000,0,(1112-0),2017,연립다세대,26.10~28.10,신규,,,
1,2026,11545,금천구,10100,가산동,1,대지,0142,0018,10.0,...,3000,73,(142-18),2025,연립다세대,26.08~28.08,신규,,,
2,2026,11680,강남구,10400,청담동,1,대지,0018,0033,2.0,...,24000,0,(18-33),2011,연립다세대,26.09~28.09,신규,,,
3,2026,11215,광진구,10100,중곡동,1,대지,0259,0006,2.0,...,1000,53,(259-6),2014,연립다세대,26.10~28.09,신규,,,
4,2026,11710,송파구,10500,석촌동,1,대지,0278,0006,3.0,...,30000,27,(278-6),2004,연립다세대,26.10~28.10,신규,,,


API 키와 같은 보안 데이터는 코드에 직접 하드코딩하지 않고 관리하는 것이 안전합니다. 
https://www.youtube.com/shorts/8OfxQ4TsYXI
- .env 파일과 python-dotenv 사용 (가장 추천 ⭐️)
프로젝트 폴더 내에 환경 변수 전용 파일(.env)을 만들고, 코드에서는 파일의 값을 읽어오는 방식입니다.

1. 패키지 설치 - 터미널(또는 주피터 셀)에서 아래 명령어를 실행합니다.

In [3]:
! pip install python-dotenv

2. .env 파일 생성 - 노트북과 같은 폴더(또는 프로젝트 최상위 폴더)에 .env 이름으로 파일을 만들고 인증키를 저장합니다.

*SEOUL_API_KEY=여기에_실제_발급받은_인증키_입력*

3. Python 코드에서 불러오기

In [4]:
import os
import requests
import pandas as pd
from dotenv import load_dotenv
# .env 파일 로드
load_dotenv()
# 환경 변수에서 API 키 읽기
API_KEY = os.getenv("SEOUL_API_KEY")
# API 호출
url = f"http://openapi.seoul.go.kr:8088/{API_KEY}/json/tbLnOpendataRentV/1/5/"
response = requests.get(url)
df = pd.DataFrame(response.json()['tbLnOpendataRentV']['row'])
df


,RCPT_YR,CGG_CD,CGG_NM,STDG_CD,STDG_NM,LOTNO_SE,LOTNO_SE_NM,MNO,SNO,FLR,...,GRFE,RTFE,BLDG_NM,ARCH_YR,BLDG_USG,CTRT_PRD,NEW_UPDT_YN,CTRT_UPDT_USE_YN,BFR_GRFE,BFR_RTFE
0,2026,11260,중랑구,10100,면목동,1,대지,1112,0000,5.0,...,11000,0,(1112-0),2017,연립다세대,26.10~28.10,신규,,,
1,2026,11545,금천구,10100,가산동,1,대지,0142,0018,10.0,...,3000,73,(142-18),2025,연립다세대,26.08~28.08,신규,,,
2,2026,11680,강남구,10400,청담동,1,대지,0018,0033,2.0,...,24000,0,(18-33),2011,연립다세대,26.09~28.09,신규,,,
3,2026,11215,광진구,10100,중곡동,1,대지,0259,0006,2.0,...,1000,53,(259-6),2014,연립다세대,26.10~28.09,신규,,,
4,2026,11710,송파구,10500,석촌동,1,대지,0278,0006,3.0,...,30000,27,(278-6),2004,연립다세대,26.10~28.10,신규,,,


In [5]:
import os
import requests
import pandas as pd
from dotenv import load_dotenv
# .env 파일 로드
load_dotenv()
# 환경 변수에서 API 키 읽기
API_KEY = os.getenv("SEOUL_API_KEY")
TYPE = "json"
SERVICE = "tbLnOpendataRentV"
all_rows = []
start_idx = 1
step = 1000
max_data = 5000  # 수집할 총 데이터 건수 예시
while start_idx <= max_data:
    end_idx = start_idx + step - 1
    url = f"http://openapi.seoul.go.kr:8088/{API_KEY}/{TYPE}/{SERVICE}/{start_idx}/{end_idx}/"
    
    res = requests.get(url)
    data = res.json()
    
    if SERVICE in data:
        rows = data[SERVICE]['row']
        all_rows.extend(rows)
        start_idx += step
    else:
        # 오류 또는 데이터 없음 시 종료
        print("결과 없음 또는 오류 발생:", data.get('RESULT', {}).get('MESSAGE'))
        break
df_total = pd.DataFrame(all_rows)
print(f"총 {len(df_total)}건 수집 완료")
df_total.head()


총 5000건 수집 완료


,RCPT_YR,CGG_CD,CGG_NM,STDG_CD,STDG_NM,LOTNO_SE,LOTNO_SE_NM,MNO,SNO,FLR,...,GRFE,RTFE,BLDG_NM,ARCH_YR,BLDG_USG,CTRT_PRD,NEW_UPDT_YN,CTRT_UPDT_USE_YN,BFR_GRFE,BFR_RTFE
0,2026,11260,중랑구,10100,면목동,1,대지,1112,0000,5.0,...,11000,0,(1112-0),2017,연립다세대,26.10~28.10,신규,,,
1,2026,11545,금천구,10100,가산동,1,대지,0142,0018,10.0,...,3000,73,(142-18),2025,연립다세대,26.08~28.08,신규,,,
2,2026,11680,강남구,10400,청담동,1,대지,0018,0033,2.0,...,24000,0,(18-33),2011,연립다세대,26.09~28.09,신규,,,
3,2026,11215,광진구,10100,중곡동,1,대지,0259,0006,2.0,...,1000,53,(259-6),2014,연립다세대,26.10~28.09,신규,,,
4,2026,11710,송파구,10500,석촌동,1,대지,0278,0006,3.0,...,30000,27,(278-6),2004,연립다세대,26.10~28.10,신규,,,


접수연도가 2026년인 데이터만 1,000건씩 분할하여 전체 수집할 수 있도록 수정

In [6]:
import os
import requests
import pandas as pd
from dotenv import load_dotenv
# 1. .env 파일 로드 및 API 키 읽기
load_dotenv()
API_KEY = os.getenv("SEOUL_API_KEY")
# 2. API 기본 정보 및 수집 조건 설정
TYPE = "json"
SERVICE = "tbLnOpendataRentV"
RCPT_YR = "2026"  # 접수연도 2026년 조건 추가
all_rows = []
start_idx = 1
step = 1000
total_count = None
print(f"[{RCPT_YR}년 서울시 부동산 전월세가 데이터 수집 시작]")
# 3. 데이터 반복 수집 (Paging)
while True:
    end_idx = start_idx + step - 1
    
    # URL 맨 뒤에 {RCPT_YR} 파라미터 추가
    url = f"http://openapi.seoul.go.kr:8088/{API_KEY}/{TYPE}/{SERVICE}/{start_idx}/{end_idx}/{RCPT_YR}/"
    
    res = requests.get(url)
    data = res.json()
    
    if SERVICE in data:
        # 첫 호출 시 2026년 전체 데이터 건수 자동 감지
        if total_count is None:
            total_count = data[SERVICE]['list_total_count']
            print(f"총 수집 대상 데이터: {total_count:,}건")
        
        rows = data[SERVICE]['row']
        all_rows.extend(rows)
        print(f"수집 진행 중: {len(all_rows):,} / {total_count:,} 건 완료")
        
        # 다음 1,000건으로 인덱스 이동
        start_idx += step
        
        # 전체 데이터 건수 이상 수집 시 루프 종료
        if len(all_rows) >= total_count:
            break
    else:
        # 오류 또는 데이터가 더 이상 없는 경우 종료
        msg = data.get('RESULT', {}).get('MESSAGE', '데이터 없음')
        print(f"수집 종료/중단: {msg}")
        break
# 4. DataFrame 변환 및 확인
df_total = pd.DataFrame(all_rows)
print(f"\n✅ 최종 {len(df_total):,}건 수집 완료!")
df_total.head()

[2026년 서울시 부동산 전월세가 데이터 수집 시작]
총 수집 대상 데이터: 392,077건
수집 진행 중: 1,000 / 392,077 건 완료
수집 진행 중: 2,000 / 392,077 건 완료
수집 진행 중: 3,000 / 392,077 건 완료
수집 진행 중: 4,000 / 392,077 건 완료
수집 진행 중: 5,000 / 392,077 건 완료
수집 진행 중: 6,000 / 392,077 건 완료
수집 진행 중: 7,000 / 392,077 건 완료
수집 진행 중: 8,000 / 392,077 건 완료
수집 진행 중: 9,000 / 392,077 건 완료
수집 진행 중: 10,000 / 392,077 건 완료
수집 진행 중: 11,000 / 392,077 건 완료
수집 진행 중: 12,000 / 392,077 건 완료
수집 진행 중: 13,000 / 392,077 건 완료
수집 진행 중: 14,000 / 392,077 건 완료
수집 진행 중: 15,000 / 392,077 건 완료
수집 진행 중: 16,000 / 392,077 건 완료
수집 진행 중: 17,000 / 392,077 건 완료
수집 진행 중: 18,000 / 392,077 건 완료
수집 진행 중: 19,000 / 392,077 건 완료
수집 진행 중: 20,000 / 392,077 건 완료
수집 진행 중: 21,000 / 392,077 건 완료
수집 진행 중: 22,000 / 392,077 건 완료
수집 진행 중: 23,000 / 392,077 건 완료
수집 진행 중: 24,000 / 392,077 건 완료
수집 진행 중: 25,000 / 392,077 건 완료
수집 진행 중: 26,000 / 392,077 건 완료
수집 진행 중: 27,000 / 392,077 건 완료
수집 진행 중: 28,000 / 392,077 건 완료
수집 진행 중: 29,000 / 392,077 건 완료
수집 진행 중: 30,000 / 392,077 건 완료
수집 진행 중: 31,000 / 392,077 

,RCPT_YR,CGG_CD,CGG_NM,STDG_CD,STDG_NM,LOTNO_SE,LOTNO_SE_NM,MNO,SNO,FLR,...,GRFE,RTFE,BLDG_NM,ARCH_YR,BLDG_USG,CTRT_PRD,NEW_UPDT_YN,CTRT_UPDT_USE_YN,BFR_GRFE,BFR_RTFE
0,2026,11710,송파구,11200,오금동,1,대지,0617,0000,20.0,...,10808,16,송파레미니스2단지,2018,아파트,26.09~28.08,갱신,,6883,38
1,2026,11620,관악구,10100,봉천동,1,대지,0100,0197,1.0,...,26700,0,동성아트빌,2003,연립다세대,26.09~28.09,갱신,○,25800,
2,2026,11110,종로구,18700,무악동,1,대지,0082,0000,11.0,...,50000,70,현대,2000,아파트,26.09~28.09,갱신,○,50000,60
3,2026,11440,마포구,12300,망원동,1,대지,0506,0000,8.0,...,56700,0,탑건진선미,2004,아파트,26.10~28.09,갱신,○,54000,
4,2026,11620,관악구,10200,신림동,1,대지,1644,0003,11.0,...,18000,51,유벤투스240,2025,아파트,26.09~28.09,신규,,,


In [7]:
df_total

,RCPT_YR,CGG_CD,CGG_NM,STDG_CD,STDG_NM,LOTNO_SE,LOTNO_SE_NM,MNO,SNO,FLR,...,GRFE,RTFE,BLDG_NM,ARCH_YR,BLDG_USG,CTRT_PRD,NEW_UPDT_YN,CTRT_UPDT_USE_YN,BFR_GRFE,BFR_RTFE
0,2026,11710,송파구,11200,오금동,1,대지,0617,0000,20.0,...,10808,16,송파레미니스2단지,2018,아파트,26.09~28.08,갱신,,6883,38
1,2026,11620,관악구,10100,봉천동,1,대지,0100,0197,1.0,...,26700,0,동성아트빌,2003,연립다세대,26.09~28.09,갱신,○,25800,
2,2026,11110,종로구,18700,무악동,1,대지,0082,0000,11.0,...,50000,70,현대,2000,아파트,26.09~28.09,갱신,○,50000,60
3,2026,11440,마포구,12300,망원동,1,대지,0506,0000,8.0,...,56700,0,탑건진선미,2004,아파트,26.10~28.09,갱신,○,54000,
4,2026,11620,관악구,10200,신림동,1,대지,1644,0003,11.0,...,18000,51,유벤투스240,2025,아파트,26.09~28.09,신규,,,
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
392072,2026,11290,성북구,13300,정릉동,1,대지,0289,0017,-1.0,...,14900,0,지산빌라,1992,연립다세대,25.11~27.11,갱신,,15000,
392073,2026,11590,동작구,10700,사당동,,,,,0.0,...,2000,83,,1993,단독다가구,25.12~26.12,신규,,,
392074,2026,11440,마포구,12600,중동,1,대지,0205,0000,2.0,...,22000,0,마이빌(205-0),2003,연립다세대,26.01~28.01,신규,,,
392075,2026,11500,강서구,10600,내발산동,1,대지,0749,0000,7.0,...,6539,12,마곡수명산파크5단지,2008,아파트,25.12~27.11,신규,,,


In [8]:
df_total.to_csv("data/seoul_rent_2026.csv", index=False, encoding="utf-8-sig")

In [9]:
# 만역 여기까지 못따라 왔다면 아래 코드로 서울시 2026년 전월세 데이터를 불러오시오.
seoul_rent_2026 = pd.read_csv("data/seoul_rent_2026.csv", encoding="utf-8-sig")
seoul_rent_2026

,RCPT_YR,CGG_CD,CGG_NM,STDG_CD,STDG_NM,LOTNO_SE,LOTNO_SE_NM,MNO,SNO,FLR,...,GRFE,RTFE,BLDG_NM,ARCH_YR,BLDG_USG,CTRT_PRD,NEW_UPDT_YN,CTRT_UPDT_USE_YN,BFR_GRFE,BFR_RTFE
0,2026,11710,송파구,11200,오금동,1.0,대지,617.0,0.0,20.0,...,10808,16,송파레미니스2단지,2018.0,아파트,26.09~28.08,갱신,NaN,6883.0,38.0
1,2026,11620,관악구,10100,봉천동,1.0,대지,100.0,197.0,1.0,...,26700,0,동성아트빌,2003.0,연립다세대,26.09~28.09,갱신,○,25800.0,NaN
2,2026,11110,종로구,18700,무악동,1.0,대지,82.0,0.0,11.0,...,50000,70,현대,2000.0,아파트,26.09~28.09,갱신,○,50000.0,60.0
3,2026,11440,마포구,12300,망원동,1.0,대지,506.0,0.0,8.0,...,56700,0,탑건진선미,2004.0,아파트,26.10~28.09,갱신,○,54000.0,NaN
4,2026,11620,관악구,10200,신림동,1.0,대지,1644.0,3.0,11.0,...,18000,51,유벤투스240,2025.0,아파트,26.09~28.09,신규,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
392072,2026,11290,성북구,13300,정릉동,1.0,대지,289.0,17.0,-1.0,...,14900,0,지산빌라,1992.0,연립다세대,25.11~27.11,갱신,NaN,15000.0,NaN
392073,2026,11590,동작구,10700,사당동,NaN,NaN,NaN,NaN,0.0,...,2000,83,NaN,1993.0,단독다가구,25.12~26.12,신규,NaN,NaN,NaN
392074,2026,11440,마포구,12600,중동,1.0,대지,205.0,0.0,2.0,...,22000,0,마이빌(205-0),2003.0,연립다세대,26.01~28.01,신규,NaN,NaN,NaN
392075,2026,11500,강서구,10600,내발산동,1.0,대지,749.0,0.0,7.0,...,6539,12,마곡수명산파크5단지,2008.0,아파트,25.12~27.11,신규,NaN,NaN,NaN


### 📋 서울시 부동산 전월세가 정보 변수 정의 (`tbLnOpendataRentV`)

#### 1. 주요 변수 목록 표 (Total 23 Columns)

| No | 변수명 (Column) | 한글 설명 | 데이터 예시 | 비고 |
| :-: | :--- | :--- | :--- | :--- |
| 1 | `RCPT_YR` | 접수연도 | `2026` | YYYY |
| 2 | `CGG_CD` | 자치구코드 | `11620` | 5자리 행정구역 코드 |
| 3 | `CGG_NM` | 자치구명 | `관악구` | 서울시 25개 자치구 |
| 4 | `STDG_CD` | 법정동코드 | `10200` | 5자리 법정동 코드 |
| 5 | `STDG_NM` | 법정동명 | `신림동` | 동 명칭 |
| 6 | `LOTNO_SE` | 지번구분 코드 | `1` | 1:대지, 2:산, 3:블럭 |
| 7 | `LOTNO_SE_NM` | 지번구분명 | `대지` | 구분 명칭 |
| 8 | `MNO` | 본번 | `0102` | 4자리 정수 |
| 9 | `SNO` | 부번 | `0024` | 4자리 정수 |
| 10 | `FLR` | 층수 | `1.0`, `5.0` | 해당 매물 층 |
| 11 | `CTRT_DAY` | 계약일 | `20260807` | YYYYMMDD |
| 12 | `RENT_SE` | 전월세 구분 | `전세`, `월세` | 거래 유형 |
| 13 | `RENT_AREA` | 임대면적 | `57.55` | 단위: $m^2$ |
| 14 | `GRFE` | 보증금 | `18000` | 단위: 만원 |
| 15 | `RTFE` | 임대료 | `100` | 단위: 만원 (전세는 `0`) |
| 16 | `BLDG_NM` | 건물명 | `(102-24)` | 단지/건물 명칭 |
| 17 | `ARCH_YR` | 건축년도 | `2002` | 준공 연도 (YYYY) |
| 18 | `BLDG_USG` | 건물용도 | `연립다세대` | 아파트/오피스텔/연립다세대/단독다가구 |
| 19 | `CTRT_PRD` | 계약기간 | `26.09~28.09` | YY.MM~YY.MM |
| 20 | `NEW_UPDT_YN` | 신규갱신여부 | `신규`, `갱신` | 계약 종류 |
| 21 | `CTRT_UPDT_USE_YN` | 계약갱신권사용여부 | `○`, ` ` | 갱신청구권 사용 여부 |
| 22 | `BFR_GRFE` | 종전 보증금 | `5000` | 갱신 계약 시 이전 보증금 (만원) |
| 23 | `BFR_RTFE` | 종전 임대료 | `88` | 갱신 계약 시 이전 임대료 (만원) |

---

#### 2. 그룹별 변수 개요
* **행정/위치 정보**: `CGG_NM`(자치구명), `STDG_NM`(법정동명), `MNO`/`SNO`(지번)
* **거래/가격 정보**: `RENT_SE`(전/월세), `GRFE`(보증금), `RTFE`(월세), `CTRT_DAY`(계약일)
* **건물/물건 정보**: `BLDG_USG`(건물용도), `BLDG_NM`(건물명), `RENT_AREA`(면적), `FLR`(층수), `ARCH_YR`(건축년도)
* **갱신 계약 정보**: `NEW_UPDT_YN`(신규/갱신), `CTRT_UPDT_USE_YN`(갱신권사용), `BFR_GRFE`/`BFR_RTFE`(종전가)


In [10]:
# 전세만 골라내기
seoul_rent_2026_jeonse = seoul_rent_2026[seoul_rent_2026["RENT_SE"] == "전세"]
seoul_rent_2026_jeonse

,RCPT_YR,CGG_CD,CGG_NM,STDG_CD,STDG_NM,LOTNO_SE,LOTNO_SE_NM,MNO,SNO,FLR,...,GRFE,RTFE,BLDG_NM,ARCH_YR,BLDG_USG,CTRT_PRD,NEW_UPDT_YN,CTRT_UPDT_USE_YN,BFR_GRFE,BFR_RTFE
1,2026,11620,관악구,10100,봉천동,1.0,대지,100.0,197.0,1.0,...,26700,0,동성아트빌,2003.0,연립다세대,26.09~28.09,갱신,○,25800.0,NaN
3,2026,11440,마포구,12300,망원동,1.0,대지,506.0,0.0,8.0,...,56700,0,탑건진선미,2004.0,아파트,26.10~28.09,갱신,○,54000.0,NaN
5,2026,11560,영등포구,10500,영등포동4가,1.0,대지,117.0,7.0,8.0,...,21700,0,스퀘어뷰,2016.0,연립다세대,26.09~28.09,갱신,NaN,20100.0,NaN
12,2026,11380,은평구,10600,대조동,1.0,대지,212.0,57.0,2.0,...,15500,0,태양빌라(212-57),1994.0,연립다세대,26.08~27.02,신규,NaN,NaN,NaN
14,2026,11500,강서구,10900,방화동,1.0,대지,817.0,0.0,3.0,...,39900,0,동성,1993.0,아파트,26.08~28.08,갱신,○,38000.0,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
392066,2026,11230,동대문구,10400,전농동,1.0,대지,690.0,0.0,7.0,...,68000,0,래미안크레시티,2014.0,아파트,26.01~28.01,신규,NaN,NaN,NaN
392070,2026,11680,강남구,10600,대치동,1.0,대지,511.0,0.0,8.0,...,108000,0,한보미도맨션2,1985.0,아파트,26.01~28.01,신규,NaN,NaN,NaN
392071,2026,11320,도봉구,10500,쌍문동,NaN,NaN,NaN,NaN,0.0,...,10000,0,NaN,2013.0,단독다가구,25.11~27.11,신규,NaN,NaN,NaN
392072,2026,11290,성북구,13300,정릉동,1.0,대지,289.0,17.0,-1.0,...,14900,0,지산빌라,1992.0,연립다세대,25.11~27.11,갱신,NaN,15000.0,NaN


In [11]:
# 동별/유형별 평균 전세가
jeonse_mean_df = seoul_rent_2026_jeonse.groupby(['CGG_NM', 'STDG_NM', 'BLDG_USG'])['GRFE'].agg(
    평균전세가_만원 = 'mean',
    거래건수 = 'count'
).round(1).reset_index()

jeonse_mean_df 

,CGG_NM,STDG_NM,BLDG_USG,평균전세가_만원,거래건수
0,강남구,개포동,단독다가구,21951.2,40
1,강남구,개포동,아파트,90199.0,830
2,강남구,개포동,연립다세대,33702.8,273
3,강남구,개포동,오피스텔,24917.2,34
4,강남구,논현동,단독다가구,21461.8,105
...,...,...,...,...,...
1184,중랑구,신내동,오피스텔,21522.4,45
1185,중랑구,중화동,단독다가구,13304.7,118
1186,중랑구,중화동,아파트,45039.5,85
1187,중랑구,중화동,연립다세대,22254.8,247


In [12]:
# 평균 전세가 기준 내림차순 정렬 (상위 10개 출력)
jeonse_mean_df = jeonse_mean_df.sort_values(by='평균전세가_만원', ascending=False)
jeonse_mean_df.head(10)

,CGG_NM,STDG_NM,BLDG_USG,평균전세가_만원,거래건수
1130,중구,장충동1가,아파트,450000.0,1
1035,종로구,신문로2가,아파트,280000.0,1
1159,중구,회현동2가,아파트,181500.0,4
925,용산구,한남동,아파트,166131.1,111
921,용산구,한강로3가,아파트,157213.1,92
479,서초구,반포동,아파트,154899.9,757
876,용산구,용산동5가,아파트,153959.7,31
405,마포구,하중동,아파트,140170.0,20
991,종로구,내수동,아파트,132300.0,15
1156,중구,회현동1가,아파트,131500.0,12


In [13]:
jeonse_mean_df.to_csv("data/jeonse_mean.csv", index=False, encoding="utf-8-sig")

## 과제2 동별/주택유형별 전세가 평균 확인 앱 만들기

### 옵션1 : 전처리된 jeonse_mean.csv 활용
#### "jeonse_mean.csv"를 이용해서 사용자가 "행정동"과 "주택유형"을 선택하면 평균 전세가를 알려주는 앱을 만들거야
- GitHub에 "seoul_rent" 리파지토리 생성
- 깃헙 "seoul_rent" 리파지토리에 모든 코드를 올리고, cloudflare로 발행
- 디자인은 심플하고, 밝은 배경에 포인트 칼라로 오렌지를 사용해줘

### 옵션2 : 열린데이터광장에서 API로 데이터를 바로 끌어와 만들기
#### 서울 열린데이터 광장에서 API로 데이터를 가져와 이용해서 사용자가 "행정동"과 "주택유형"을 선택하면 평균 전세가를 알려주는 앱을 만들거야
- GitHub에 "seoul_rent_api" 리파지토리 생성
- 깃헙 "seoul_rent_api" 리파지토리에 모든 코드를 올리고, cloudflare로 발행
- 디자인은 심플하고, 밝은 배경에 포인트 칼라로 오렌지를 사용해줘
- 자치구와 행정동을 선택하면, 선택된 행정동의 2026년 데이터를 모두 가져와줘
- 내 API키는 .env에 있어. API키가 노출되지 않도록 보안 설정해줘




샘플 URL
샘플 URL	서울시 부동산 전월세가 정보
http://openapi.seoul.go.kr:8088/(인증키)/xml/tbLnOpendataRentV/1/5/
예제	

요청인자
변수명	타입	변수설명	값설명
KEY	String(필수)	인증키	OpenAPI 에서 발급된 인증키
TYPE	String(필수)	요청파일타입	xml : xml, xml파일 : xmlf, 엑셀파일 : xls, json파일 : json
SERVICE	String(필수)	서비스명	tbLnOpendataRentV
START_INDEX	INTEGER(필수)	요청시작위치	정수 입력 (페이징 시작번호 입니다 : 데이터 행 시작번호)
END_INDEX	INTEGER(필수)	요청종료위치	정수 입력 (페이징 끝번호 입니다 : 데이터 행 끝번호)
RCPT_YR	STRING(선택)	접수연도	YYYY
CGG_CD	STRING(선택)	자치구코드	5자리 정수
CGG_NM	STRING(선택)	자치구명	문자열
STDG_CD	STRING(선택)	법정동코드	5자리 정수
LOTNO_SE	STRING(선택)	지번구분	1:대지,2:산,3:블럭
MNO	STRING(선택)	본번	4자리 정수
SNO	STRING(선택)	부번	4자리 정수
CTRT_DAY	STRING(선택)	계약일	YYYYMMDD
BLDG_NM	STRING(선택)	건물명	문자열
BLDG_USG	STRING(선택)	건물용도	아파트/단독다가구/연립다세대/오피스텔 택1
출력값
No	출력명	출력설명
공통	list_total_count	총 데이터 건수 (정상조회 시 출력됨)
공통	RESULT.CODE	요청결과 코드 (하단 메세지설명 참고)
공통	RESULT.MESSAGE	요청결과 메시지 (하단 메세지설명 참고)
1	RCPT_YR	접수연도
2	CGG_CD	자치구코드
3	CGG_NM	자치구명
4	STDG_CD	법정동코드
5	STDG_NM	법정동명
6	LOTNO_SE	지번구분
7	LOTNO_SE_NM	지번구분명
8	MNO	본번
9	SNO	부번
10	FLR	층
11	CTRT_DAY	계약일
12	RENT_SE	전월세 구분
13	RENT_AREA	임대면적(㎡)
14	GRFE	보증금(만원)
15	RTFE	임대료(만원)
16	BLDG_NM	건물명
17	ARCH_YR	건축년도
18	BLDG_USG	건물용도
19	CTRT_PRD	계약기간
20	NEW_UPDT_YN	신규갱신여부
21	CTRT_UPDT_USE_YN	계약갱신권사용여부
22	BFR_GRFE	종전 보증금
23	BFR_RTFE	종전 임대료

## Cloudflare Pages 발행 (소요 시간: 약 1분)

### 1️⃣ Cloudflare 프로젝트 생성
1. **[Cloudflare Dashboard](https://dash.cloudflare.com/)**에 로그인 후 **`Workers & Pages`** 메뉴로 이동합니다.
2. **`Create application`** $\rightarrow$ **`Pages`** 탭 선택 후 **`[Connect to Git]`** 버튼을 클릭합니다.
3. 저장소 목록에서 **`jslee-gses/seoul_rent_api`** 저장소를 선택합니다.

### 2️⃣ 빌드 설정 (Build Settings)
| 설정 항목 | 입력 값 |
| :--- | :--- |
| **Framework preset** | `None` |
| **Build command** | *(빈칸)* |
| **Build output directory** | `/` |

---

### 3️⃣ API Key 환경 변수 (Environment Variables) 등록
프로젝트 생성 완료 후, **`Settings`** $\rightarrow$ **`Environment variables`** 메뉴로 이동하여 아래 변수를 등록합니다.

* **Variable name**: `SEOUL_API_KEY`
* **Value**: `본인의 API키 입력`

> 💡 **완료**: 환경 변수 저장 후 **`Deployments`** 메뉴에서 **[Retry deployment]** (재배포)를 클릭하시면 서비스가 최종 등록됩니다!
